unit_tests/aero/UnitTestFSIturbine.C

```diff
@@ -57,4 +206,74 @@ TEST_F(CylinderMesh, construct_FSIturbine)
   EXPECT_NO_THROW(fsiTurb.setup(bulk));
 }
 
+TEST_F(CylinderMesh, construct_OpenfastFSI)
+{
+  if (stk::parallel_machine_size(MPI_COMM_WORLD) != 1) {
+    GTEST_SKIP();
+  }
+
+  const double innerRadius = 1.0;
+  const double outerRadius = 2.0;
+  fill_mesh_and_initialize_test_fields(20, 20, 20, innerRadius, outerRadius);
+
+  YAML::Node yamlNode = create_openfastFSI_yaml_node();
+  sierra::nalu::OpenfastFSI openfastFSI(yamlNode);
+  const double dtNalu = 6.25e-3;
+  EXPECT_NO_THROW(openfastFSI.setup(dtNalu, bulk));
+  EXPECT_NO_THROW(openfastFSI.end_openfast());
+}
+
+TEST_F(CylinderMesh, call_fsiTurbine_mapLoads)
+{
+  if (stk::parallel_machine_size(MPI_COMM_WORLD) != 1) {
+    GTEST_SKIP();
+  }
+
+  const double innerRadius = 1.0;
+  const double outerRadius = 2.0;
+  fill_mesh_and_initialize_test_fields(20, 20, 20, innerRadius, outerRadius);
+
+  YAML::Node yamlNode = create_openfastFSI_yaml_node();
+  sierra::nalu::OpenfastFSI openfastFSI(yamlNode);
+  const double dtNalu = 6.25e-3;
+  EXPECT_NO_THROW(openfastFSI.setup(dtNalu, bulk));
+  std::cout << "nTurbinesGlob: " << openfastFSI.get_nTurbinesGlob()
+            << std::endl;
+  EXPECT_EQ(1, openfastFSI.get_nTurbinesGlob());
+
+  const int turbIndex = 0;
+  sierra::nalu::fsiTurbine* fsiTurb = openfastFSI.get_fsiTurbineData(turbIndex);
+  EXPECT_TRUE(fsiTurb != nullptr);
+  fast::turbineDataType& params = fsiTurb->params_;
+
+  EXPECT_EQ(0, params.nBRfsiPtsTwr);
+  EXPECT_EQ(0, params.numBlades);
+
+  set_tower_ref_pos(*bulk, *fsiTurb);
+
+  fsiTurb->computeLoadMapping();
+
+  // stk::mesh::field_fill(0.1, *tforceSCS_);
+  set_x_mesh_disp(*bulk, 0.1);
+  set_x_values(fsiTurb->brFSIdata_.twr_def, 0.1);
+
+  fsiTurb->mapLoads();
+
+  constexpr double expectedEndLoad = -120;
```

Comment:
How is this computed? I must be missing it, but I don't see where the loads are set or computed in the unit test fixture.

Reply:
Yes it took me quite a while to figure out how to make it produce non-zero loads.
fsiTurbine::mapLoads calls `calc_loads_->execute()` which computes the loads. I was able to make it produce nice round numbers like -120, -240, by setting fields like viscosity, density etc to all ones in the unit-test fixture.

Reply:
ah interesting